lib/datasets/rdatasets.rb

```diff
@@ -0,0 +1,109 @@
+require_relative "dataset"
+require_relative "tar_gz_readable"
+
+module Datasets
+  class RdatasetsList < Dataset
+    Record = Struct.new(:package,
+                        :dataset,
+                        :title,
+                        :rows,
+                        :cols,
+                        :n_binary,
+                        :n_character,
+                        :n_factor,
+                        :n_logical,
+                        :n_numeric,
+                        :csv,
+                        :doc)
+
+    def initialize
+      super
+      @metadata.id = "rdatasets"
+      @metadata.name = "Rdatasets"
+      @metadata.url = "https://vincentarelbundock.github.io/Rdatasets/"
+      @metadata.licenses = ["GPL-3"]
+      @data_url = "https://raw.githubusercontent.com/vincentarelbundock/Rdatasets/master/datasets.csv"
+      @data_path = cache_dir_path + "datasets.csv"
+    end
+
+    def download(force: false)
+      if force || !@data_path.exist?
+        super(@data_path, @data_url)
+      end
+    end
+
+    def dataset_info(package_name, dataset_name)
+      each_dataset(package_name) do |row|
+        if row["Item"] == dataset_name
+          record = row.to_h
+          record.transform_keys! {|key| key.downcase.to_sym }
+          return record
+        end
+      end
+    end
+
+    def each(package_name = nil)
+      return to_enum(__method__, package_name) unless block_given?
+
+      download
+      CSV.open(@data_path, headers: :first_row, converters: :all) do |csv|
+        csv.each do |row|
+          if package_name.nil? || row["Package"] == package_name
+            yield Record.new(*row.fields)
+          end
+        end
+      end
+    end
+  end
+
+  class Rdatasets < Dataset
+    def self.exist?(package_name, dataset_name = nil)
+      list = RdatsetsList.new
+      if dataset_name.nil?
+        list.each(package_name).any?
+      else
+        list.each(package_name).any? {|r| r.dataset == dataset_name }
+      end
+    end
+
+    def initialize(package_name, dataset_name)
+      list = RdatasetsList.new
+
+      info = list.each(package_name).find {|r| r.dataset == dataset_name }
+      unless info
+        raise ArgumentError, "Unable to locate dataset #{package_name}/#{dataset_name}"
+      end
+
+      super()
+      @metadata.id = "rdatasets-#{package_name}-#{dataset_name}"
+      @metadata.name = "Rdatasets: #{package_name}: #{dataset_name}"
+      @metadata.url = info.csv
+      @metadata.licenses = ["GPL-3"]
+      @metadata.description = info.title
+
+      # Follow the original directory structure in the cache directory
+      @data_path = list.send(:cache_dir_path) + "csv" + package_name + (dataset_name + ".csv")
```

Comment:
Should we share cache directory of `RdatasetsList`?
If we have separated cache directory, we can clear cache for each data by `Datasets::Dataset#clear_cache`.